plugins/outputs/mongodb/mongodb.go

```diff
@@ -119,6 +121,48 @@ func (s *MongoDB) Init() error {
 		username.Destroy()
 		password.Destroy()
 		s.clientOptions.SetAuth(credential)
+	case "PLAIN":
+		if s.Username.Empty() {
+			return errors.New("authentication for PLAIN must specify a username")
+		}
+		if s.Password.Empty() {
+			return errors.New("authentication for PLAIN must specify a password")
+		}
+		username, err := s.Username.Get()
+		if err != nil {
+			return fmt.Errorf("getting username failed: %w", err)
+		}
+		password, err := s.Password.Get()
+		if err != nil {
+			username.Destroy()
+			return fmt.Errorf("getting password failed: %w", err)
+		}
+		credential := options.Credential{
+			AuthMechanism: "PLAIN",
+			AuthSource:    "$external",
+			Username:      username.String(),
+			Password:      password.String(),
+		}
+		username.Destroy()
+		password.Destroy()
```

Comment:
How about

```suggestion
		usernameRaw, err := s.Username.Get()
		if err != nil {
			return fmt.Errorf("getting username failed: %w", err)
		}
		username := usernameRaw.String(),
		username.Destroy()
		
		passwordRaw, err := s.Password.Get()
		if err != nil {
			return fmt.Errorf("getting password failed: %w", err)
		}
		password := passwordRaw.String()
		password.Destroy()
		
		credential := options.Credential{
			AuthMechanism: "PLAIN",
			AuthSource:    "$external",
			Username:      username,
			Password:      password,
		}
```

which is a bit more robust I think...